WcaOnRails/app/jobs/sync_mailing_lists_job.rb

```diff
@@ -6,7 +6,7 @@ class SyncMailingListsJob < ApplicationJob
   def perform
     GsuiteMailingLists.sync_group("delegates@worldcubeassociation.org", User.delegates.map(&:email))
     GsuiteMailingLists.sync_group("seniors@worldcubeassociation.org", User.senior_delegates.map(&:email))
-    GsuiteMailingLists.sync_group("leaders@worldcubeassociation.org", TeamMember.current.where(team_leader: true).map(&:user).map(&:email))
+    GsuiteMailingLists.sync_group("leaders@worldcubeassociation.org", TeamMember.current.where(team_id: Team.official).where(team_leader: true).map(&:user).map(&:email))
```

Comment:
Given how often I've seen the `where(team_id: Team.official)` and the other `where` clauses, I think it would make sense to create scopes on the `TeamMember` model.
That way if we ever change our mind about what is "official" or "team leader" or "team senior member" we just have to change the scope and not all the places it's used.

To be more accurate I suggest to add the following to the `TeamMember` model:
```ruby
# I don't think you need the map(&:id) below, but I may be wrong
scope :in_official_team, -> { where(team_id: Team.all_official) }
scope :leader, -> { where(team_leader: true) }
scope :senior_member, -> { where(team_senior_member: true) }
```

This way I think you can get rid of the `official` scope on the `Team` model, which actually looks like very redundant considering the `all_official` method (as far as I understand it, they both return the same set of objects)!

Reply:
That sounds good to me, although maybe it should be handled on a separate PR?

Reply:
I think it's fine to do it here, the grep for the use of these `where`s is pretty short:
```
app/jobs/sync_mailing_lists_job.rb:9:    GsuiteMailingLists.sync_group("leaders@worldcubeassociation.org", TeamMember.current.where(team_leader: true).map(&:user).map(&:email))
app/jobs/sync_mailing_lists_job.rb:17:    GsuiteMailingLists.sync_group("treasurer@worldcubeassociation.org", Team.wfc.current_members.where(team_leader: true).map(&:user).map(&:email))
app/models/user.rb:34:    team_leaders = TeamMember.current.where(team_leader: true).map(&:user)
app/models/user.rb:35:    team_senior_members = TeamMember.current.where(team_senior_member: true).map(&:user)
app/models/user.rb:43:    team_leaders = TeamMember.current.where(team_leader: true).map(&:user)
app/views/teams/index.html.erb:24:              <%= users_to_sentence team.current_members.where(team_leader: true).map(&:user) %>
```

Reply:
I've taken care of this on cd5e74a1f9db429570fadaf2142fa40eb3dc6481. I added the `.map(&:id)` not because it was necessary, but because it made sense to me given the preceding `team_id:`.

Reply:
I added back the `official` scope on the Team model as it seems to still be useful for the `foo_of_any_official_team?` methods.